lib/web_ui/lib/src/engine/configuration.dart

```diff
@@ -51,14 +52,49 @@ void debugSetConfiguration(FlutterConfiguration configuration) {
 class FlutterConfiguration {
   /// Constructs a configuration from a JavaScript object containing
   /// runtime-supplied properties.
-  FlutterConfiguration(this._js);
+  FlutterConfiguration(JsFlutterConfiguration? config) {
+    addConfigurationOverrides(config);
+  }
 
-  final JsFlutterConfiguration? _js;
+  final List<JsFlutterConfiguration> _configurations = <JsFlutterConfiguration>[];
+
+  /// Adds a configuration [overrides] object that will take precedence over
+  /// any existing configuration.
+  ///
+  /// This is used so we support multiple sources of configuration with a consistent
+  /// precedence. From lowest to highest:
+  ///
+  /// * default static values
+  /// * window.flutterConfiguration object
+  /// * engineInitializer.initEngine(runtimeConfig)
+  ///
+  /// Overrides are always added at position [0] of the array, so the order in
+  /// which this method is called (first with flutterConfiguration), then with
+  /// the runtimeConfig, matters.
+  void addConfigurationOverrides(JsFlutterConfiguration? overrides) {
+    if (overrides != null) {
+      _configurations.insert(0, overrides);
+    }
+  }
+
+  T getConfigValue<T>(T? Function(JsFlutterConfiguration) extractor, T defaultValue) {
+    try {
```

Comment:
The try/catch here is unfortunate. Could you instead do something like:
```
return _configurations.map(extractor).firstWhere((T value) => value != null, () => defaultValue);
```

Reply:
This is gone now. Ideally I should have used [this `firstWhereOrNull`](https://api.flutter.dev/flutter/package-collection_collection/IterableExtension/firstWhereOrNull.html) that does exactly what I needed (but I wasn't sure if the engine was already depending on the collection package)